src/app/review/page.tsx

```diff
@@ -1,36 +1,20 @@
-import { Form, NumberField, Select, TextField } from '@/components/Form';
+import { AddReviewForm } from '@/components/Form';
 import { getRestaurants } from '@/services/restaurants';
-import { addReview } from '@/services/review';
-import { Restaurant } from '@/types/generated';
-import { use } from 'react';
 
-const ReviewSelect = (props: { restaurants: Promise<Restaurant[]> }) => {
-  'use client';
-  const restaurants = use(props.restaurants);
-  return (
-    <Select id="restaurant" label="Restaurant" name="restaurant">
-      {restaurants.map((restaurant) => (
-        <option key={restaurant.id} value={restaurant.id}>
-          {restaurant.name}
-        </option>
-      ))}
-    </Select>
-  );
-};
-
-const ReviewPage = () => {
-  const restaurants = getRestaurants();
+const ReviewPage = async () => {
+  const restaurants = await getRestaurants();
+  const restaurantOptions = [
```

Comment:
tää pitäis mennä mun mielestä promisena tonne AddReviewFormille ja siel käyttää sitä use hookkia. Tän vois esim kombottaa tohon edelliseen ilman awaittia vaikka .then:nillä?

Reply:
https://react.dev/reference/react/use

```
- When fetching data in a [Server Component](https://react.dev/reference/rsc/use-server), prefer async and await over use. async and await pick up rendering from the point where await was invoked, whereas use re-renders the component after the data is resolved.

- Prefer creating Promises in [Server Components](https://react.dev/reference/rsc/use-server) and passing them to [Client Components](https://react.dev/reference/rsc/use-client) over creating Promises in Client Components. Promises created in Client Components are recreated on every render. Promises passed from a Server Component to a Client Component are stable across re-renders. [See this example](https://react.dev/reference/react/use#streaming-data-from-server-to-client).
```

Reply:
using await in a [Server Component](https://react.dev/reference/react/components#server-components) will block its rendering until the await statement is finished. Passing a Promise from a Server Component to a Client Component prevents the Promise from blocking the rendering of the Server Component.